src/main/java/jenkins/scm/api/SCMFileSystem.java

```diff
@@ -483,6 +552,25 @@ public final boolean supports(SCMSourceDescriptor descriptor) {
          */
         protected abstract boolean supportsDescriptor(SCMSourceDescriptor descriptor);
 
+        /**
+         * Given a {@link SCM} this should try to build a corresponding {@link SCMFileSystem} instance that
+         * reflects the content at the specified {@link SCMRevision}. If the {@link SCM} is supported but not
+         * for a fixed revision, best effort is acceptable as the most capable {@link SCMFileSystem} will be returned
+         * to the caller.
+         *
+         * @param build the build of the {@link SCM}
+         * @param scm the {@link SCM}.
+         * @param rev the specified {@link SCMRevision}.
+         * @return the corresponding {@link SCMFileSystem} or {@code null} if this builder cannot create a {@link
+         * SCMFileSystem} for the specified {@link SCM}.
+         * @throws IOException          if the attempt to create a {@link SCMFileSystem} failed due to an IO error
+         *                              (such as the remote system being unavailable)
+         * @throws InterruptedException if the attempt to create a {@link SCMFileSystem} was interrupted.
+         */
+        @CheckForNull
+        public abstract SCMFileSystem build(@NonNull Run build, @NonNull SCM scm, @CheckForNull SCMRevision rev)
```

Comment:
I think a default with 
```java
   build(build.getParent(), scm, rev);
```

Would be none breaking and backwards compatible with anyone who have not switched to overriding build the build run.

Than [`build(@NonNull Item owner, @NonNull SCM scm, @CheckForNull SCMRevision rev)`](https://github.com/tgatinea/scm-api-plugin/blob/acbaed50f624409d04f810ca466b426ef58d68ff/src/main/java/jenkins/scm/api/SCMFileSystem.java#L589-L591) could marked deprecated

Reply:
I was just thinking to update my 3 PRs with the last version of plugins (I currently uses them in production but they are getting old now) when I saw your comments.
@dwnusbaum I fully understand that the scm-api-plugin is widely used and that any change can break other plugin. I'm too newbe to get the big picture with all involvement.
That's why I appreciate some help.
@jetersen I understand that adding "return build(build.getParent(), scm, rev);" would keep bacward compatibility. 
I'm not familiar with the java syntax <?, ?> meaning. Anyway I've taken into account your remarks and pushed your code